search/utils.py

```diff
@@ -33,104 +33,132 @@ def index_indicator(self, indicator):
 
         data = self.get_field_data(indicator)
 
-        # aggregate related models
-        data['workflowlevel1'] = list(map(lambda w: w.name, indicator.workflowlevel1.all()))
-
-        # index data with elasticsearch
-        try:
-            self.es.index(index=self.prefix+"indicators", id=data['id'], doc_type='indicator', body=data)
-        except RequestError:
-            print(indicator, "Error")   # Todo write to error log
-
-    def delete_indicator(self, id):
+        for wf1 in indicator.workflowlevel1.all():
+            if wf1.organization is not None and wf1.organization is not None:
+                org_uuid = str(wf1.organization.organization_uuid)
+
+                # aggregate related models
+                data['workflowlevel1'] = list(map(lambda w: w.name, [wf1]))
+                # data['workflowlevel1'] = list(map(lambda w: w.name, indicator.workflowlevel1.all()))
+
+                # index data with elasticsearch
+                try:
+                    self.es.index(index=self.prefix + org_uuid + "_indicators", id=data['id'], doc_type='indicator',
+                             body=data)
+                except RequestError:
+                    print(indicator, "Error")   # Todo write to error log
+            else:
+                continue
+
+    def delete_indicator(self, indicator):
         if self.es is None:
             return
         try:
-            self.es.delete(index=self.prefix + "indicators", id=id, doc_type='indicator')
+            for wf1 in indicator.workflowlevel1.all():
```

Comment:
This is not really efficient DB-wise. Get just the values that you need with `values_list()`.

Reply:
You forgot to do this change. If you can't use `values_list()`, use `select_related()`